nuxeo-runtime/nuxeo-runtime-stream/src/test/java/org/nuxeo/runtime/stream/tests/TestStreamService.java

```diff
@@ -157,4 +161,29 @@ public void testDisabledStreamProcessor() throws Exception {
         }
     }
 
+
+    @Test
+    public void testProbe() throws Exception {
+        @SuppressWarnings("resource")
+        StreamManager streamManager = service.getStreamManager("default");
+        StreamProbe probe = new StreamProbe();
+        probe.reset();
+        ProbeStatus status = probe.run();
+        assertFalse(status.isFailure());
+        // generate a failure
+        try {
+            streamManager.append("inputFailure", Record.of("key", null));
+            Thread.sleep(500);
+            // the probe failure is delayed by 1s after the first probe run
+            status = probe.run();
+            assertFalse(status.toString(), status.isFailure());
+
+            Thread.sleep(1500);
+            status = probe.run();
+            assertTrue(status.toString(), status.isFailure());
```

Comment:
You could probably use [Awaitility](https://github.com/awaitility/awaitility) here, something like:
```
import static org.awaitility.Awaitility.await;
...
await().atLeast(500, MILLISECONDS).until(!probe.run().isFailure())
await().atLeast(1500, MILLISECONDS).until(probe.run().isFailure())
```
Yet, not sure how you could handle the assertion error messages.

Reply:
If I remember well, this is the first method (await) which can take the message.

Reply:
interesting, but here I think it makes the test harder to read